R/case_weights.R

```diff
@@ -79,36 +80,42 @@ NULL
 #'
 #' @export
 #' @name case-weight-helpers
-get_case_weights <- function(info, .data) {
+get_case_weights <- function(info, .data, call = rlang::caller_env()) {
   wt_col <- info$variable[info$role == "case_weights" & !is.na(info$role)]
 
-
   if (length(wt_col) == 1) {
     res <- .data[[wt_col]]
     if (!is.numeric(res)) {
-      rlang::abort(
-        paste0(
-          "Column ", wt_col, " has a 'case_weights' role but is not numeric."
-        )
+      cli::cli_abort(
+        c(
+          "x" = "{.arg {wt_col}} has a {.code case_weights} role,\\
+                 but is not numeric.",
+          "i" = "{.arg {wt_col}} is {.obj_type_friendly {wt_col}}."
+          ),
+        call = call
```

Comment:
tried for like 15 minutes, and was not able to reach this abort.

It is not the worst thing to keep i think? since it is just guarded behind a `is.numeric()` which is fast.

Reply:
All good, thanks for the note. :)